patches/server/1065-Fire-and-handle-canellable-TickRateChangedEvent.patch

```diff
@@ -0,0 +1,51 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Axionize <154778082+Axionize@users.noreply.github.com>
+Date: Tue, 8 Oct 2024 20:32:34 -0400
+Subject: [PATCH] Fire and handle canellable TickRateChangedEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/ServerTickRateManager.java b/src/main/java/net/minecraft/server/ServerTickRateManager.java
+index 37dcf3dc3e50afd85912a7496c828576a38a4e9c..765c17532d13b136c51fe68a970aab36b8fb7f24 100644
+--- a/src/main/java/net/minecraft/server/ServerTickRateManager.java
++++ b/src/main/java/net/minecraft/server/ServerTickRateManager.java
+@@ -121,12 +121,22 @@ public class ServerTickRateManager extends TickRateManager {
+         this.sprintTimeSpend += System.nanoTime() - this.sprintTickStartTime;
+     }
+ 
++    // Paper start - Fire and handle cancellable TickRateChangeEvent
++    public void setTickRate(org.bukkit.command.CommandSender commandSender, float tickRate) {
++        org.bukkit.event.server.TickRateChangeEvent event = new org.bukkit.event.server.TickRateChangeEvent(commandSender, tickRate);
++        event.callEvent();
++        if (!event.isCancelled()) {
++            super.setTickRate(tickRate);
++            this.server.onTickRateChanged();
++            this.updateStateToClients();
++        }
++    }
++
+     @Override
+     public void setTickRate(float tickRate) {
+-        super.setTickRate(tickRate);
+-        this.server.onTickRateChanged();
+-        this.updateStateToClients();
++        setTickRate(null, tickRate);
```

Comment:
This diff can be minimized by moving the override down and doing an early return instead of an if(notCancelled).